scripts/lib/basic_youtube_upload.py

```diff
@@ -130,20 +151,84 @@ def resumable_upload(request):
 
+def get_known_playlist(title):
+    return PLAYLIST_IDS[title]
+
+def get_playlist(youtube, title):
+    """Return users's playlist ID by title (None if not found)"""
+    playlists = youtube.playlists()
+    request = playlists.list(mine=True, part="id,snippet")
+    current_encoding = locale.getpreferredencoding()
+    
+    while request:
+        results = request.execute()
+        for item in results["items"]:
+            t = item.get("snippet", {}).get("title")
+            existing_playlist_title = (t.encode(current_encoding) if hasattr(t, 'decode') else t)
+            if existing_playlist_title == title:
+                return item.get("id")
+        request = playlists.list_next(request, results)
+
+def create_playlist(youtube, title, privacy):
+    """Create a playlist by title and return its ID"""
+    debug("Creating playlist: {0}".format(title))
+    response = youtube.playlists().insert(part="snippet,status", body={
+        "snippet": {
+            "title": title,
+        },
+        "status": {
+            "privacyStatus": privacy,
+        }
+    }).execute()
+    return response.get("id")
+
+def add_video_to_existing_playlist(youtube, playlist_id, video_id):
+    """Add video to playlist (by identifier) and return the playlist ID."""
+    debug("Adding video to playlist: {0}".format(playlist_id))
+
+    body = {
+        "snippet": {
+            "playlistId": playlist_id,
+            "position": 0,
+            "resourceId": {
+                "kind": "youtube#video",
+                "videoId": video_id,
+            }
+        }
+    }
 
-def upload_video(path, **kwargs):
-    youtube = get_youtube_client()
     try:
-        initialize_upload(youtube, file=path, **kwargs)
-    except HttpError as e:
-        print(f'An HTTP error {e.resp.status} occurred:\n{e.content}')
\ No newline at end of file
+        return youtube.playlistItems().insert(part="snippet",
+                                              body=body).execute()
+    except HttpError as error:
+        parsed = parse_youtube_http_error(error)
+        # A "manualSortRequired" error means the playlist is not manually
+        # sorted, and it needs to be in order to use "position". So just try
+        # again without setting "position" this time. (It's dumb, but the API
+        # appears to provide no way to test for this ahead of time.)
+        if parsed and any(error["reason"] == "manualSortRequired" for error in parsed["errors"]):
+            del body["snippet"]["position"]
+            return youtube.playlistItems().insert(part="snippet",
+                                                  body=body).execute()
+        else:
+            raise
+
+def add_video_to_playlist(youtube, video_id, title, privacy="unlisted"):
+    """Add video to playlist (by title) and return the full response."""
+    playlist_id = get_known_playlist(title) or \
+        get_playlist(youtube, title) or \
+        create_playlist(youtube, title, privacy)
```

Comment:
FYI, if you wrap this expression in parens, you can drop the trailing `\`:

```suggestion
    playlist_id = (get_known_playlist(title) or
                   get_playlist(youtube, title) or
                   create_playlist(youtube, title, privacy))
```

Both ways are totally fine, I just happen those think the parentheses make for *slightly* nicer style. Totally personal preference.